base/compiler/ssair/irinterp.jl

```diff
@@ -216,17 +218,18 @@ function reprocess_instruction!(interp::AbstractInterpreter,
                     ir.stmts[idx][:flag] |= IR_FLAG_EFFECT_FREE
                 end
             end
-        elseif inst.head === :invoke
+        elseif head === :invoke
             mi′ = inst.args[1]::MethodInstance
             if mi′ !== irsv.mi # prevent infinite loop
                 rt = concrete_eval_invoke(interp, inst, mi′, irsv)
             end
-        elseif inst.head === :throw_undef_if_not
-            # TODO: Terminate interpretation early if known false?
+        elseif head === :throw_undef_if_not || # TODO: Terminate interpretation early if known false?
+               head === :gc_preserve_begin ||
+               head === :gc_preserve_end
             return false
         else
             ccall(:jl_, Cvoid, (Any,), inst)
```

Comment:
Should normally use `Core.println` in the compiler. The `jl_()` function is normally intended for things that are likely to explode, and gdb.